addons/mrp/models/mrp_unbuild.py

```diff
@@ -84,11 +84,18 @@ def _onchange_company_id(self):
     @api.onchange('mo_id')
     def _onchange_mo_id(self):
         if self.mo_id:
+            if self.lot_id and self.lot_id not in self.mo_id.move_finished_ids.move_line_ids.lot_id:
+                raise UserError(_("The selected serial number does not correspond to the one used in the manufacturing order, please select another one."))
             self.product_id = self.mo_id.product_id.id
             self.product_qty = self.mo_id.product_qty
             self.product_uom_id = self.mo_id.product_uom_id
             self.bom_id = self.mo_id.bom_id
 
+    @api.onchange('lot_id')
+    def _onchange_lot_id(self):
+        if self.lot_id and self.mo_id and self.lot_id not in self.mo_id.move_finished_ids.move_line_ids.lot_id:
+            raise UserError(_("The selected serial number does not correspond to the one used in the manufacturing order, please select another one."))
```

Comment:
Instead of duplicate the code, it could be done only in `_onchange_mo_id` if you add `lot_id` in the `@api.onchange`, no ?

Reply:
Yes it works, I wanted to do that, but if we want to add specific code to lot_id in the future it's better to separate the two onchange